core/objects.go

```diff
@@ -523,6 +524,10 @@ func (cert Certificate) MatchesCSR(csr *x509.CertificateRequest, earliestExpiry
 		err = InternalServerError("Generated certificate doesn't have correct key usage extensions")
 		return
 	}
+	if len(parsedCertificate.PolicyIdentifiers) == 0 {
+		err = InternalServerError("Generated certificate doesn't have any policy identifiers")
+		return
+	}
```

Comment:
This is incorrect.  Ideally, that they're the same set as configured, but I would settle for checking that the number of policy identifiers is the same.